### PR TITLE
feat(modal): add tokens and update shim

### PR DESCRIPTION
**Pull request by dtsanevmw**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: projects/angular/src/modal/_properties.modal.scss [hunk 1]
```diff
@@ -19,14 +24,27 @@
...
+      --clr-modal-content-border-width: 0;
+      --clr-modal-content-border-radius: unset;
+      --clr-modal-content-padding: #{variables.$clr_baselineRem_1};
 
       // Usage: ../modal/_modal.clarity.scss
       --clr-modal-title-color: var(--clr-h3-color);
       --clr-modal-title-font-family: var(--clr-h3-font-family);
       --clr-modal-title-font-weight: var(--clr-h3-font-weight);
+      --clr-modal-header-padding: 0 0 #{variables.$clr_baselineRem_1} 0;
+      --clr-modal-footer-padding: #{variables.$clr_baselineRem_1} 0 0 0;
```
Comment (valentin-mladenov) [inline projects/angular/src/modal/_properties.modal.scss:46]: Shouldn't that be like original state?

```suggestion
      --clr-modal-footer-padding: #{variables.$clr_baselineRem_0_5 variables.$clr_baselineRem_1 0 0};
```

File: projects/angular/src/modal/_properties.modal.scss [hunk 2]
```diff
@@ -19,14 +24,27 @@
...
+      --clr-modal-content-border-color: transparent;
+      --clr-modal-content-border-width: 0;
+      --clr-modal-content-border-radius: unset;
+      --clr-modal-content-padding: #{variables.$clr_baselineRem_1};
 
       // Usage: ../modal/_modal.clarity.scss
       --clr-modal-title-color: var(--clr-h3-color);
       --clr-modal-title-font-family: var(--clr-h3-font-family);
       --clr-modal-title-font-weight: var(--clr-h3-font-weight);
+      --clr-modal-header-padding: 0 0 #{variables.$clr_baselineRem_1} 0;
```
Comment (valentin-mladenov) [inline projects/angular/src/modal/_properties.modal.scss:45]: Shouldn't that be like original state?

```suggestion
      --clr-modal-header-padding: #{0 variables.$clr_baselineRem_1 variables.$clr_baselineRem_0_5 0};
```

File: projects/angular/src/modal/_properties.modal.scss [hunk 3]
```diff
@@ -19,14 +24,27 @@
...
+        #{variables.$clr_baselineRem_2px}
+        var(--clr-modal-content-box-shadow-color);
       --clr-modal-backdrop-color: var(--clr-color-neutral-900);
       --clr-modal-backdrop-opacity: 0.85;
       --clr-modal-border-radius: var(--clr-global-borderradius);
+      --clr-modal-content-border-style: none;
+      --clr-modal-content-border-color: transparent;
+      --clr-modal-content-border-width: 0;
+      --clr-modal-content-border-radius: unset;
+      --clr-modal-content-padding: #{variables.$clr_baselineRem_1};
```
Comment (valentin-mladenov) [inline projects/angular/src/modal/_properties.modal.scss:39]: Shouldn't that be like original state?

```suggestion
      --clr-modal-content-padding: #{variables.$clr_baselineRem_0_5 0 variables.$clr_baselineRem_0_5 variables.$clr_baselineRem_1};
```

File: projects/angular/src/modal/_modal.clarity.scss [hunk 4]
```diff
@@ -168,11 +218,22 @@
...
+      gap,
+      clr-modal-footer-gap,
+      variables.$clr_baselineRem_0_5,
+      variables.$clr-use-custom-properties
+    );
 
     .btn {
       //switched right margin to left because footer is right aligned
-      margin: 0 0 0 variables.$clr_baselineRem_0_5;
+      margin: 0;
```
Comment (valentin-mladenov) [inline projects/angular/src/modal/_modal.clarity.scss:236]: If I understood this correctly you are removing between button margins and replacing them with gaps. Right?

File: projects/angular/src/modal/_modal.clarity.scss [hunk 5]
```diff
@@ -168,11 +218,22 @@
   .modal-footer {
     display: flex;
     justify-content: flex-end;
-    padding: variables.$clr_baselineRem_1 0 0 0;
+    @include mixins.css-var(
+      padding,
+      clr-modal-footer-padding,
+      variables.$clr_baselineRem_1 0 0 0,
+      variables.$clr-use-custom-properties
+    );
```
Comment (valentin-mladenov) [inline projects/angular/src/modal/_modal.clarity.scss:226]: `clr-modal-footer-padding` have more than one possible value.

1. `variables.$clr_baselineRem_1 0 0 0`
2. `variables.$clr_baselineRem_0_5 variables.$clr_baselineRem_1 0 0`

Try using different variables.

File: projects/ui/src/shim.cds-core.scss [hunk 6]
```diff
@@ -536,9 +536,23 @@
...
+  --clr-modal-content-border-width: var(--cds-alias-object-border-width-100);
+  --clr-modal-content-border-radius: var(--cds-alias-object-border-radius-100);
+  --clr-modal-content-padding: var(--cds-global-space-9);
+  --clr-modal-header-padding: 0 0 var(--cds-global-space-9) 0;
+  --clr-modal-footer-padding: var(--cds-global-space-9) 0 0 0;
+  --clr-modal-footer-gap: var(--cds-global-space-5);
+  --clr-modal-sm-width: calc(288 * (1rem / var(--clr-global-font-size)));
+  --clr-modal-md-width: calc(576 * (1rem / var(--clr-global-font-size)));
+  --clr-modal-lg-width: calc(864 * (1rem / var(--clr-global-font-size)));
+  --clr-modal-xl-width: calc(1152 * (1rem / var(--clr-global-font-size)));
```
Comment (valentin-mladenov) [inline projects/ui/src/shim.cds-core.scss:555]: The design says the modal `height` should be 248px.
I don't see anything related to it. 
We should have some `min-height` somewhere, probably?
I don't see anything in core as well.

**Review by valentin-mladenov — CHANGES_REQUESTED** [review]
Check my comments.

File: projects/angular/src/modal/_properties.modal.scss (hunk 1 above)
Comment (dtsanevmw) [reply projects/angular/src/modal/_properties.modal.scss:46]: This was the Ng state -`padding: variables.$clr_baselineRem_1 0 0 0;`

File: projects/angular/src/modal/_properties.modal.scss (hunk 2 above)
Comment (dtsanevmw) [reply projects/angular/src/modal/_properties.modal.scss:45]: That's the value of the padding on smaller screens, so no. I'll add separate properties for small variant.

File: projects/angular/src/modal/_properties.modal.scss (hunk 3 above)
Comment (dtsanevmw) [reply projects/angular/src/modal/_properties.modal.scss:39]: Same thing here, what you are referring is the small variant.

File: projects/angular/src/modal/_modal.clarity.scss (hunk 4 above)
Comment (dtsanevmw) [reply projects/angular/src/modal/_modal.clarity.scss:236]: Yeah, it's pretty much the same.

**Review by valentin-mladenov — CHANGES_REQUESTED** [review]
I see an extra border around the wizard content, are you sure it should be there?

image

**Comment by dtsanevmw** [comment]
The NG light theme difference is correct and it was mistaken due to not imported properties in the modals and missing visual tests before.

**Review by valentin-mladenov — APPROVED** [review]
LGTM

**Pull request merged**